### PR TITLE
Compile filters in call node

### LINKED ISSUE #76: Calling macro with a parameter invoking a filter crashes
So I have a macro
`{% macro someMacro someParameter %}`

And when I attempt to invoke it like this:
`{% call someMacro myType.storedVariables|annotated:"someAnnotation" %}`

I get this fatal exception:

`Terminating app due to uncaught exception 'NSUnknownKeyException', reason: '[<SourceryRuntime.Struct 0x7f94a2e1a5a0> valueForUndefinedKey:]: this class is not key value coding-compliant for the key storedVariables|annotated:"json".'`

Calling the macro without the filter on storedVariables works just fine:
`{% call someMacro myType.storedVariables %}`

### PR DESCRIPTION
**Pull request by ilyapuchka**
Resolves #76

### CODE REVIEW

File: Tests/StencilSwiftKitTests/CallNodeTests.swift [hunk 1]
```diff
@@ -38,7 +38,8 @@ class CallNodeTests: XCTestCase {
     }
 
     XCTAssertEqual(node.variableName, "myFunc")
-    XCTAssertEqual(node.arguments, [Variable("a"), Variable("b"), Variable("c")])
+    let variables = node.arguments.flatMap({ $0 as? FilterExpression }).flatMap({ $0.variable })
```
Comment (AliSoftware) [inline Tests/StencilSwiftKitTests/CallNodeTests.swift]: I generally agree with this code style too when chaining flatMaps but our linter disagrees there, so maybe adapt the syntax here, like `let variables = node.arguments.flatMap { ($0 as? FilterExpression)?.variable }`?

**Comment by AliSoftware** [comment]
I think there was another linting violation reported by the CI — but on BoolFiltersTests (which is strange because you didn't change it in that PR, maybe a leftover violation that passed though just at the time migrated to CircleCI 2 and the PR checks had to change to adapt and we missed that and let it slip, idk? would need fixing anyway so might as well fix in this PR)

**Comment by ilyapuchka** [comment]
@AliSoftware which one? I don't see errors or warnings in the lint logs

**Comment by AliSoftware** [comment]
@ilyapuchka Nvm, false alarm, my bad.

The story is that I looked at [the CircleCI report](https://circleci.com/gh/SwiftGen/StencilSwiftKit/454) from my iPhone, but since CircleCI isn't really responsive and doesn't display the logs in a super-adapted way on mobile, I got confused by the console logs. The console writes `Linting 'ParseBoolTests.swift' (6/12)` just before the first `/Users/distiller/project/Tests/StencilSwiftKitTests/CallNodeTests.swift:41:21: warning: Trailing Closure Violation: Trailing closure syntax should be used whenever possible. (trailing_closure)` so I first misread that on my tiny phone screen as the violation being the result of linting `ParseBoolTests.swift`)

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,14 @@
 
 ---
 
+## Master
+
+### Bug fixes
+
+* Fixed using filter expression in call node.
```
Comment (AliSoftware) [inline CHANGELOG.md]: Missing two spaces at the end of the line after the `.` — for markdown formatting purposes

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 ### Bug fixes
 
-* Fixed using filter expression in call node.
+* Fixed using filter expression in call node.
```
Comment (AliSoftware) [inline CHANGELOG.md]: 👍

**Pull request merged**